New Pint version causes test failures

* xclim version: master
* Python version: 3.8
* Operating System: linux

### Description
<!--Describe what you were trying to get done.
Tell us what happened, what went wrong, and what you expected to happen.-->

Builds in PR #467 are currently failing on `TestPrecipAccumulation`. They were not failing on my laptop until I updated pint from 0.11 to 0.12 and now I can reproduce the issue. 
Pint 0.12 was released at 11:43 today.

`TestPrecipAccumulation.test_3d_data_with_nans` is failing with traceback:
```
___________________________________ TestPrecipAccumulation.test_3d_data_with_nans ____________________________________

self = <test_precip.TestPrecipAccumulation object at 0x7fd154327280>

    def test_3d_data_with_nans(self):
        # test with 3d data
        pr = xr.open_dataset(self.nc_pr).pr  # mm/s
        prMM = xr.open_dataset(self.nc_pr).pr
        prMM *= 86400
        prMM.attrs["units"] = "mm/day"
        # put a nan somewhere
        prMM.values[10, 1, 0] = np.nan
        pr.values[10, 1, 0] = np.nan
    
        out1 = atmos.precip_accumulation(pr, freq="MS")
        out2 = atmos.precip_accumulation(prMM, freq="MS")
    
        # test kg m-2 s-1
        pr.attrs["units"] = "kg m-2 s-1"
        out3 = atmos.precip_accumulation(pr, freq="MS")
    
        np.testing.assert_array_almost_equal(out1, out2, 3)
>       np.testing.assert_array_almost_equal(out1, out3)
E       AssertionError: 
E       Arrays are not almost equal to 6 decimals
E       
E       Mismatched elements: 58580 / 157248 (37.3%)
E       Max absolute difference: 1.49309635e-05
E       Max relative difference: 5.91187429e-08
E        x: array([[[ 79.099993,  78.069995,  76.660005, ...,  72.889999,
E                 71.969998,  70.510003],
E               [       nan,  78.829994,  77.469998, ...,  73.349994,...
E        y: array([[[ 79.09999 ,  78.06999 ,  76.66    , ...,  72.89    ,
E                 71.97    ,  70.51    ],
E               [       nan,  78.829994,  77.47    , ...,  73.34999 ,...

tests/test_precip.py:86: AssertionError
```

--- a/tests/test_precip.py
+++ b/tests/test_precip.py
@@ -83,7 +83,7 @@ class TestPrecipAccumulation:
         out3 = atmos.precip_accumulation(pr, freq="MS")
 
         np.testing.assert_array_almost_equal(out1, out2, 3)
-        np.testing.assert_array_almost_equal(out1, out3)
+        np.testing.assert_array_almost_equal(out1, out3, 5)
 
         # check some vector with and without a nan
         x1 = prMM[:31, 0, 0].values

